sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Extensions/ServiceRequestExtensions.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.WebPubSub.AspNetCore
+{
+    /// <summary>
+    /// Help methods to parse upstream requests.
+    /// </summary>
+    public static class ServiceRequestExtensions
```

Comment:
Do all of these methods need to be public? Some of them are on very generic types (`HttpRequest/Dictionary<string, object>/string`) so they would appear to customers in scenarios not related to WPS

Reply:
The public ones are for WPS when customer doesn't want to implement `ServiceHub` want to read `HttpRequest` and parse by themselves. Under this assumption, I put a few as public.

Reply:
The public ones are for WPS when customer doesn't want to implement `ServiceHub` want to read `HttpRequest` and parse by themselves. Under this assumption, I put a few as public.